packages/x-components/src/x-modules/tagging/service/__tests__/pdpd-add-to-cart.service.spec.ts

```diff
@@ -0,0 +1,124 @@
+import { InMemoryStorageService } from '@empathyco/x-storage-service';
+import { Store } from 'vuex';
+import { DefaultPDPAddToCartService } from '../pdp-add-to-cart.service';
+import { XPlugin } from '../../../../plugins/index';
+import { installNewXPlugin } from '../../../../__tests__/utils';
+import { taggingXModule } from '../../x-module';
+import { RootXStoreState } from '../../../../store/index';
+import { createResultStub } from '../../../../__stubs__/index';
+
+interface PDPdAddToCartServiceTestAPI {
+  service: DefaultPDPAddToCartService;
+  localStorageService: InMemoryStorageService;
+  sessionStorageService: InMemoryStorageService;
+  store: Store<RootXStoreState>;
+}
+
+function preparePDPAddToCartService(): PDPdAddToCartServiceTestAPI {
+  XPlugin.resetInstance();
+  installNewXPlugin();
+  XPlugin.registerXModule(taggingXModule);
+
+  const store = XPlugin.store;
+  const localStorageService = new InMemoryStorageService();
+  const sessionStorageService = new InMemoryStorageService();
+  const service = new DefaultPDPAddToCartService(localStorageService, sessionStorageService);
+
+  return {
+    service,
+    localStorageService,
+    sessionStorageService,
+    store
+  };
+}
+
+describe('testing pdp add to cart', () => {
+  const { service, localStorageService, sessionStorageService, store } =
+    preparePDPAddToCartService();
+
+  const result = createResultStub('Product 001');
+
+  const localGetItemSpy = jest.spyOn(localStorageService, 'getItem');
+  const localSetItemSpy = jest.spyOn(localStorageService, 'setItem');
+  const localRemoveItemSpy = jest.spyOn(localStorageService, 'removeItem');
+
+  const sessionGetItemSpy = jest.spyOn(sessionStorageService, 'getItem');
+  const sessionSetItemSpy = jest.spyOn(sessionStorageService, 'setItem');
+  const storeDispatchSpy = jest.spyOn(store, 'dispatch');
+
+  const url = 'http://localhost:8080/?param=test&param2=&param3=t+e+s+t';
+  Object.defineProperty(window, 'location', {
+    writable: true,
+    value: new URL(url)
+  });
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    localStorageService.clear();
+    sessionStorageService.clear();
+  });
+
+  it('stores the result with the id as key and ttl', () => {
+    store.commit('x/tagging/setClickedResultStorageKey', 'id');
```

Comment:
Can you extract these two lines to a function? It's repeated a couple of times in the tests. Maybe making TTL optional too.